mlflow/recipes/steps/train.py

```diff
@@ -398,16 +435,22 @@ def my_warn(*args, **kwargs):
 
             target_data = raw_validation_df[self.target_col]
             prediction_result = model.predict(raw_validation_df.drop(self.target_col, axis=1))
-            if self.predict_proba:
-                prediction_result_probs = model.predict_proba(
-                    raw_validation_df.drop(self.target_col, axis=1)
+
+            use_predict_proba = isinstance(prediction_result, pd.DataFrame)
+            if use_predict_proba:
```

Comment:
Isn't it possible for the wrapped model to output a DataFrame even if the model doesn't have a `classes_` attribute? Is there a more specific check we can use?

Reply:
Currently the wrapped model would only output a DF if it has `classes_` or else it would output the label directly rather than a df with probabilities, label and score. 
I couldn't find a better alternative for the same. Have any suggestions?

Reply:
What if I pass in a dataframe with multiple rows as an input to self._classifier.predict()? The output wouldn’t be a scalar, right? Would it be a list? Numpy array? Are we certain that it’s not a DataFrame?